src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -482,9 +481,11 @@ private void OnDocEnded(List<ReplicatedDocument> replications, bool pushing)
                             Config.Database.ResolveConflict(docID, Config.ConflictResolver);
                             replication = replication.ClearError();
                         } catch (Exception e) {
+                            replication = replication.AssignError(new C4Error(C4ErrorCode.NotFound));
```

Comment:
This is going to assign "NotFound" to any exception thrown unconditionally.  See what happens now if you try to verify the errors on other kinds of errors like "db from different document", etc